discussion/app/discussion/DiscussionApi.scala

```diff
@@ -34,21 +40,24 @@ trait DiscussionApi extends Http with ExecutionContexts with Logging {
   }
 
   def commentFor(id: Int, displayThreaded: Option[String]  = None): Future[Comment] = {
-
-    val url = s"$apiRoot/comment/$id?displayResponses=true" +
-      displayThreaded.map{ dt => "&displayThreaded=" + dt  }.getOrElse("")
+    val parameters = List(
+      "displayResponses" -> "true",
+      "displayThreaded" -> displayThreaded)
+    val url = endpointUrl(s"/comment/$id", parameters)
     getCommentJsonForId(id, url)
   }
 
   def commentsFor(key: DiscussionKey, params: DiscussionParams): Future[DiscussionComments] = {
-    // displayThreaded=true can return an error on old discussions.
-    val url = s"$apiRoot/discussion/$key" + (if(params.topComments) "/topcomments" else "")+
-                    s"""|?pageSize=${params.pageSize}
-                    |&page=${params.page}
-                    |&orderBy=${params.orderBy}
-                    |${if(params.displayThreaded) "" else "&displayThreaded=false"}
-                    |&showSwitches=true""".stripMargin.replace("\n", "")+
-                    params.maxResponses.map{i => "&maxResponses="+ i}.getOrElse("")
+    val parameters = List(
+      "pageSize" -> params.pageSize,
+      "page" -> params.page,
+      "orderBy" -> params.orderBy,
+      "displayThreaded" -> (params.displayThreaded match {
+        case false => "false"
+        case _ => None}),
```

Comment:
No. DisplayThreaded won't be added to the query string if None. That's the default behavior of the NetAPorter.uri when adding query string parameters lib https://github.com/NET-A-PORTER/scala-uri/blob/3c438f082392a722e1c0913b69fe21f458de5af2/src/main/scala/com/netaporter/uri/Uri.scala#L65